src/Forms/Form.php

```diff
@@ -317,6 +320,51 @@ public function __construct(
         $this->setupDefaultClasses();
     }
 
+    /**
+     * Make the form require sudo mode, which will make the form readonly and add a sudo mode password field
+     * unless the current user previously activated sudo mode.
+     */
+    public function requireSudoMode(): void
+    {
+        // Check that the current request handler for the form is one that's used
+        // in an admin context where sudo mode makes sense
+        $classes = [
+            LeftAndMain::class,
+            GridFieldDetailForm_ItemRequest::class,
+        ];
+        $enableSudoMode = false;
+        foreach ($classes as $class) {
+            if (class_exists($class) && is_a($this->getController(), $class)) {
```

Comment:
```suggestion
            if (is_a($this->getController(), $class)) {
```
No need to check if the class exists - if it doesn't exist, `is_a` will return false, because the controller isn't that class.

Reply:
Updated